rag/svr/task_executor_refactor/dataset_wiki_generator.py

```
@@ -223,6 +223,60 @@ def _pipeline_compilation_template_ids(pipeline_id: str, tenant_id: str) -> list
+    if not pipeline_id:
+        return None
+    from api.db.services.canvas_service import UserCanvasService
+
+    ok, canvas = UserCanvasService.get_by_id(pipeline_id)
+    if not ok or not canvas:
+        return None
+    dsl = getattr(canvas, "dsl", None)
+    if isinstance(dsl, str):
+        try:
+            dsl = json.loads(dsl)
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)